### DISCUSSION TITLE
Proposal: Extending Polaris-Web for Version Compatibility and Multi-Extension Handling

### DISCUSSION THREAD

**Discussion by HunnySajid** [WebOfTrust/polaris-web#29] [Ideas]
**Current Situation:**
The polaris-web package currently interacts with compatible browser extensions but lacks mechanisms to:

- Ensure version compatibility between the package and extensions.
- Handle situations where multiple compatible extensions are installed.

**Proposed Features:**
**1- Version Compatibility:** Implement version-check functionality to detect compatible extension versions.
**2- Multiple Extensions:** Allow the detection of multiple installed extensions, enabling:
 - User selection of a preferred extension.
 - Broadcasting requests to all or specific extensions.

**Additional Use Cases:**
- Conflict resolution between multiple extensions’ responses.
- Prioritization of extensions based on features or user preference.
- Extensible support for new extensions in the future.

I’m looking for input on these ideas and any other potential use cases or challenges this might raise.

CC: @lenkan @2byrds

**Comment by 2byrds** [comment]
Great write-up @HunnySajid I can't wait to hear from @rodolfomiranda and @lenkan so we can push this ⏩

**Comment by HunnySajid** [comment]
Linked Issue: https://github.com/WebOfTrust/signify-browser-extension/issues/156

**Comment by lenkan** [comment]
Can a very first simple step be to store the `signify-extension` message on the client instance. If the client has received `signify-extension` from multiple extensions, it can reject all subsequent requests with an error message saying "Multiple signify extensions are currently not supported". At least the users will then get consistent behaviour?

**Comment by edeykholt** [comment]
I've been thinking through this and will have a set of suggestions soon.  
A few questions first:
1) Other than the extension letting itself be known to the page when it injects its content script, are any of the extension's interactions with a page *not* in response to actions the user (or a page script) initiates?  If so, what's an example?
2) Are any projects planning on not using polaris-web for interacting with signify browser extension(s)?

**Comment by edeykholt** [comment]
See refactoring proposal at https://github.com/WebOfTrust/polaris-web/discussions/31
